### ISSUE TITLE
dotnet version range syntax not supported

### ISSUE THREAD

**Issue by dmccaffery** [vscode-contrib/vscode-versionlens#58]
NuGet uses a different syntax for version ranges in that it supports scoped ranges rather than just partial matches.
Syntax:

[] = Inclusive
() = Exclusive
\* = \<any\>

For instance, the following should match versions >=1.0.0 and < 2.0.0. Since the `commandFactory` uses the semver parser from nodejs; it fails to parse ranged versions.

```xml
<PackageReference Include="Package.Name" Version="[1.0.0, 2.0.0)" />
```

Expected:

The version lens should list the following:

* latest available version within the version range
* latest available pre-release version within the version range

When either of these are selected, the left segment of the range should be updated to reflect the selected version unless the right segment is inclusive and equal to the selected version.

For example:

`[1.0.0, 2.0.0)` when `1.9.9` is available should result in a new value of `[1.9.9, 2.0.0)`
`[1.0.0, 1.9.9]` when `1.9.9` is available should result in a new value of `1.9.9` without a range.

In addition, the version lens should list the following:

* latest available version
* latest available pre-release version

When either of these are selected, the version range should be updated to match the selected version without a range.

For example:

`[1.0.0, 2.0.0)` when `2.0.1` is available should result in a new value of `2.0.1` without a range.

Actual:

`Invalid semver version entered`

**Comment by pflannery** [comment]
Pre-releases (tagged versions) will be available in v0.17.0

PRs are welcome for the nuget ranges.
For a reference to current nuget ranges https://docs.microsoft.com/en-us/nuget/create-packages/dependency-versions#version-ranges

**Comment by pflannery** [comment]
[pre-releases/0.18.0-beta.2](https://github.com/vscode-contrib/vscode-versionlens/releases/tag/0.18.0-beta.2) contains nuget range syntax support.

One caveat is that providing maximums like `[,2.0.0)` don't work yet. This is also the same with npm at the moment though.

Minimum ranges will work `<PackageReference Include="bootstrap" Version="(3,]" />` = v3.3.7 latest
Fixed is ok too `<PackageReference Include="bootstrap" Version="[3]" />` = v3.0.0

The overall strategy will need some widening to support ranges that cap to specified maximums and yet still provide the latest versions along with prereleases.

**Comment by pflannery** [comment]
Now implemented in [0.18.0](https://github.com/vscode-contrib/vscode-versionlens/blob/master/CHANGELOG.md#0180) (with the maximum range issue fixed)

![image](https://cloud.githubusercontent.com/assets/1727302/26177638/62b8f6b8-3b52-11e7-9d7b-c37754a62011.png)

and will break down ranges when no match found

![image](https://cloud.githubusercontent.com/assets/1727302/26177672/89333768-3b52-11e7-9a6f-e34c6c7db5a6.png)

handles floating ranges

![image](https://cloud.githubusercontent.com/assets/1727302/26177772/fef72b62-3b52-11e7-9938-be6c0c2f0959.png)

**Comment by dmccaffery** [comment]
You are amazing, thank you!
